src/gcp/cloudsql/permissions.ts

```diff
@@ -0,0 +1,237 @@
+import { Options } from "../../options";
+import { needProjectId } from "../../projectUtils";
+import { executeSqlCmdsAsIamUser, executeSqlCmdsAsSuperUser } from "./connect";
+import { testIamPermissions } from "../iam";
+import { logger } from "../../logger";
+import { concat } from "lodash";
+import { FirebaseError } from "../../error";
+
+export function firebaseowner(databaseId: string) {
+  return `firebaseowner_${databaseId}_public`;
+}
+
+export function firebasereader(databaseId: string) {
+  return `firebasereader_${databaseId}_public`;
+}
+
+export function firebasewriter(databaseId: string) {
+  return `firebasewriter_${databaseId}_public`;
+}
+
+// Returns true if "grantedRole" is granted to "granteeRole" and false otherwise.
+// Throw an error if commands fails due to another reason like connection issues.
+export async function checkRoleIsGranted(
+  options: Options,
+  instanceId: string,
+  databaseId: string,
+  grantedRole: string,
+  granteeRole: string,
```

Comment:
Consider using the parameter object pattern here - this is a long list of args of the same type and it would be easy for a future dev to transpose args. See go/tott/550

Reply:
I'm tempted to leave this as is since many of our existing functions take instanceId, databaseId.